src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/CrossCaseTypeSearchRequest.java

```diff
@@ -45,12 +56,75 @@ public JsonNode getSearchRequestJsonNode() {
     }
 
     public boolean isMultiCaseTypeSearch() {
-        return caseTypeIds.size() > 1;
+        return multiCaseTypeSearch;
     }
 
     private void validateJsonSearchRequest() {
         if (!searchRequestJsonNode.has(QUERY)) {
             throw new BadSearchRequest("missing required field 'query'");
         }
     }
+
+    public List<String> getSourceFilterAliasFields() {
+        return sourceFilterAliasFields;
+    }
+
+    public static class Builder {
+
+        private final List<String> caseTypeIds = new ArrayList<>();
+        private JsonNode searchRequestJsonNode;
+        private boolean multiCaseTypeSearch;
+        private final List<String> sourceFilterAliasFields = new ArrayList<>();
+
+        public Builder withCaseTypes(List<String> caseTypeIds) {
+            if (caseTypeIds != null) {
+                this.caseTypeIds.addAll(caseTypeIds);
+            }
+            return this;
+        }
+
+        public Builder withSearchRequest(JsonNode searchRequestJsonNode) {
+            this.searchRequestJsonNode = searchRequestJsonNode;
+            return this;
+        }
+
+        public Builder withMultiCaseTypeSearch(boolean multiCaseTypeSearch) {
+            this.multiCaseTypeSearch = multiCaseTypeSearch;
+            if (multiCaseTypeSearch && searchRequestJsonNode.has(SOURCE)) {
+                setSourceFilterAliasFields(searchRequestJsonNode.get(SOURCE));
+            }
+            return this;
+        }
+
+        public Builder withSourceFilterAliasFields(List<String> sourceFilterAliasFields) {
+            if (sourceFilterAliasFields != null) {
+                this.sourceFilterAliasFields.addAll(sourceFilterAliasFields);
+            }
+            return this;
+        }
+
+        private void setSourceFilterAliasFields(JsonNode multiCaseTypeSearchSourceNode) {
+            if (multiCaseTypeSearchSourceNode != null && multiCaseTypeSearchSourceNode.isArray()) {
+                // Alias fields are expected in source filter for multi-case types searches. If no aliases are defined in source filter, only meta data will
+                // be returned with no data fields. If aliases are defined in the source filter, alias fields with data and metadata will be returned.
+                sourceFilterAliasFields.addAll(StreamSupport.stream(multiCaseTypeSearchSourceNode.spliterator(), false)
```

Comment:
can we have:
`sourceFilterAliasFields.addAll(getAliasFieldsFromSource())`

Reply:
done